database/database.go

```diff
@@ -0,0 +1,246 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/AlexeyVilkov/go_final_project/date"
+	"github.com/AlexeyVilkov/go_final_project/model"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// объявляем глобальную переменную
+var Db *sqlx.DB
+
+// подключение к БД
+func ConnectDB() (*sqlx.DB, error) {
+
+	// проверяем, есть ли файл БД
+	appPath, err := os.Executable()
+	if err != nil {
+		log.Fatal(err)
+	}
+	dbFile := filepath.Join(filepath.Dir(appPath), "scheduler.db")
+	_, err = os.Stat(dbFile)
+
+	var install bool
+	if err != nil {
+		install = true
+	}
+
+	// создаём файл БД
+	if install {
+		file, err := os.Create("scheduler.db")
+		if err != nil {
+			log.Fatal(err)
+		}
+		file.Close()
+	}
+
+	// настройка подключения к БД
+	db, err := sqlx.Open("sqlite3", "scheduler.db")
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	// если БД отсутствовала, то создаём таблицу и индекс
+	if install {
+		// тексты sql-запросов
+		createTableSQL := "CREATE TABLE scheduler (id INTEGER PRIMARY KEY AUTOINCREMENT, date TEXT, title TEXT, comment TEXT, repeat TEXT)"
+		createIndexSQL := "CREATE INDEX ixDate ON scheduler (date asc)"
+
+		// создание таблицы
+		if _, err = db.Exec(createTableSQL); err != nil {
+			fmt.Println("ERROR: create table - ", err)
+			return nil, err
+		}
+
+		// создание индекса
+		if _, err = db.Exec(createIndexSQL); err != nil {
+			fmt.Println("ERROR: create index - ", err)
+			return nil, err
+		}
+
+		// вставка записи
+		/*if _, err = db.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES('20250218', 'Тест title', 'Тест comment', 'Тест repeat')"); err != nil {
+			fmt.Println("ERROR: insert into - ", err)
+			return nil, err
+		}*/
+	}
+
+	// присваиваем глобальной переменной созданную БД
+	Db = db
+
+	return db, nil
+}
+
+func PostTask(tsk model.Task) (int, error) {
+	// запись задачи в БД
+	res, err := Db.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES(:date, :title, :comment, :repeat)",
+		sql.Named("date", tsk.Date),
+		sql.Named("title", tsk.Title),
+		sql.Named("comment", tsk.Comment),
+		sql.Named("repeat", tsk.Repeat))
+	if err != nil {
+		fmt.Println("ERROR: insert into - ", err)
+		return 0, err
+	}
+
+	id, err := res.LastInsertId()
+	if err != nil {
+		fmt.Println("ERROR: не удалось получить последний id - ", err)
+		return 0, err
```

Comment:
Лучше в таких слоях приложения не делать выводы, а оборачивать ошибку и ее возвращать, а вызывающая сторона сама решит, что делать с ошибкой.
```suggestion
		return 0, fmt.Errorf("error in getting last id: %w", err)
```